src/main/drivers/pwm_output_dshot.c

```diff
@@ -96,6 +102,129 @@ void pwmWriteDshotInt(uint8_t index, uint16_t value)
     }
 }
 
+#ifdef USE_DSHOT_TELEMETRY
+uint32_t readDoneCount;
+
+static void motor_DMA_IRQReadHandler(dmaChannelDescriptor_t *descriptor)
+{
+    if (DMA_GET_FLAG_STATUS(descriptor, DMA_IT_TCIF)) {
+        motorDmaOutput_t * const motor = &dmaMotors[descriptor->userParam];
+
+        motor->hasTelemetry = true;
+        DMA_Cmd(motor->timerHardware->dmaRef, DISABLE);
+        TIM_DMACmd(motor->timerHardware->tim, motor->timerDmaSource, DISABLE);
+        DMA_CLEAR_FLAG(descriptor, DMA_IT_TCIF);
+
+        readDoneCount++;
+    }
+}
+
+static uint16_t decodeDshotPacket(uint32_t buffer[])
+{
+    int value = 0;
+    for (int i = 1; i < DSHOT_TELEMETRY_INPUT_LEN; i += 2) {
+        int diff = buffer[i] - buffer[i-1];
+        value <<= 1;
+        if (diff > 0) {
```

Comment:
So really what this is doing is a comparison after modulo difference similar to the proshot code:         
```
        const int proshotModulo = MOTOR_NIBBLE_LENGTH_PROSHOT;
        int diff = ((buffer[i] + proshotModulo - buffer[i-1]) % proshotModulo) - PROSHOT_BASE_SYMBOL;

```

I was trying to make if faster since quite a bit of time was spent there. Not sure it was worth it. I don't think I've profiled it.